src/function/scalar/string/substring.cpp

```diff
@@ -10,6 +10,29 @@
 
 namespace duckdb {
 
+static const auto SUPPORTED_UPPER_BOUND = NumericLimits<int32_t>::Maximum();
```

Comment:
Could this maybe be changed to `uint32_t` max? As that is the maximum string size that our system can store.

Reply:
Huh, I assumed we could store `numeric_limits<uint64_t>::max()` size since `string_t::GetSize()` returns a `idx_t`. Good to know, sure, I'll fix it.

Reply:
That is mostly by convention, the length is stored as a `uint32_t` internally:

```cpp
	union {
		struct {
			uint32_t length;
			char prefix[4];
			char *ptr;
		} pointer;
		struct {
			uint32_t length;
			char inlined[12];
		} inlined;
	} value;
```